### PR TITLE
Fix diamond collector metrics

### PR DESCRIPTION
**Pull request by gnufied**
Because they broke when all collectors report via
same collector channel

### CODE REVIEW

File: src/fullerite/collectors_test.go [hunk 1]
```diff
@@ -116,8 +116,13 @@ func TestReadFromCollector(t *testing.T) {
 	go func() {
 		defer wg.Done()
 		collector.Channel() <- metric.New("hello")
-		time.Sleep(time.Duration(3) * time.Second)
-		collector.Channel() <- metric.New("world")
+		time.Sleep(time.Duration(2) * time.Second)
```
Comment (avadhutp) [inline src/fullerite/collectors_test.go]: Does not block this ticket, but later on maybe we can remove the `time.Sleep`s from the test. It makes it unnecessarily long to run. We will have to mock out calls to `time.Now` in `readFromCollector` method.

But this isn't necessary for this ticket.

**Comment by avadhutp** [comment]
:ship: it :+1:

File: src/fullerite/collectors_test.go (hunk 1 above)
Comment (gnufied) [inline src/fullerite/collectors_test.go]: yeah it adds nearly 6 seconds to test times, not great. We can do one of those DI things and move this time checking to a function.

**Comment by gnufied** [comment]
I will merge this for now and open another ticket for mocking out calls to time checks.

File: src/fullerite/collectors_test.go (hunk 1 above)
Comment (avadhutp) [inline src/fullerite/collectors_test.go]: yup, something of an enhancement though.

**Comment by poros** [comment]
Could you please add a lengthier description?

**Comment by gnufied** [comment]
Yes sure. I will add one here and one in code.

**Pull request merged**